File: frontend/src/model/SchoolStatistics.ts
import {SchoolClass} from "./SchoolClass";
import {StudentView} from "./User";
import {Pair} from "./Pair";

export interface SchoolStatistics {
    success: number,
    grades: number,
    absences: number,
    feedback: number,
    examinations: number,
    events: number,
    schoolClassToAverageGrade: Pair<SchoolClass, number | null>[],
    studentToAverageGrade: Pair<StudentView, number | null>[]
}

File: backend/src/main/kotlin/com/nevexis/backend/schoolManagement/statistics/SchoolStatistics.kt
package com.nevexis.backend.schoolManagement.statistics

import com.nevexis.backend.schoolManagement.school_class.SchoolClass
import com.nevexis.backend.schoolManagement.users.StudentView
import java.math.BigDecimal

data class SchoolStatistics(
    val success: BigDecimal,
    val grades: Int,
    val absences: Int,
    val feedback: Int,
    val examinations: Int,
    val events: Int,
    val schoolClassToAverageGrade: List<Pair<SchoolClass, BigDecimal?>>,
    val studentToAverageGrade: List<Pair<StudentView, BigDecimal?>>
)


